x/programs/rust/expose_macro/src/lib.rs

```diff
@@ -17,10 +17,15 @@ pub fn expose(_: TokenStream, item: TokenStream) -> TokenStream {
     let name = &input.sig.ident;
     let input_args = &input.sig.inputs;
     let new_name = Ident::new(&format!("{}_guest", name), name.span()); // Create a new name for the generated function(name that will be called by the host)
-
-    let full_params = input_args.iter().map(|fn_arg| {
+    let empty_param = Ident::new("ctx", Span::call_site()); // Create an empty parameter for the generated function
+    let full_params = input_args.iter().enumerate().map(|(index, fn_arg)| {
         // A typed argument is a parameter. An untyped(reciever) argument is a self parameter.
         if let FnArg::Typed(PatType { pat, ty, .. }) = fn_arg {
+            // ensure first parameter is Context
+            if index == 0 && !is_context(ty) {
```

Comment:
You should additionally check if the name is `init`. Otherwise, there's nothing stopping someone from adding a `ctx` param to their other functions when it doesn't actually get injected which would probably cause UB.

Reply:
i think we want to force every function to take a `ctx` as the first param. The host will always inject this as the first parameter.

Reply:
(required change)

Reply:
whoops, my bad haha